whois-commons/src/main/java/net/ripe/db/whois/common/etree/IntervalMap.java

```diff
@@ -161,4 +161,20 @@ public interface IntervalMap<K extends Interval<?>, V> {
      *         intervals.
      */
     List<V> findExactAndAllMoreSpecific(K key);
+
+    /**
+     * Finds all values associated with intervals that are contained within (more specific than)
+     * and whose prefix is most high
+     * <code>key</code>.
+     * <p/>
+     * <p/>
+     * The resulting values are the most specific values associated with the
+     * <code>key</code>.
+     *
+     * @param key the key to find the exact and all levels more specific values
+     *            for
+     * @return the (possibly empty) list of values associated with the matching
+     *         intervals.
+     */
+    List<V> findMostSpecific(K key);
```

Comment:
Should it be list of just one element as it is most specific?

Reply:
List
there can be a lot of most specific. It is the last leaf of the tree but the tree could have more than one branch

Reply:
Ok got it we are getting the node with no children. Thanks

Reply:
I gave you an example in slack